vector/src/main/java/im/vector/riotx/features/home/room/list/RoomListFragment.kt

```diff
@@ -203,6 +215,29 @@ class RoomListFragment : VectorBaseFragment(), RoomSummaryController.Listener, O
         }
     }
 
+    private fun handleQuickActions(quickActions: RoomListQuickActions) {
+        when (quickActions) {
+            is RoomListQuickActions.NotificationsAllNoisy     -> {
+                roomListViewModel.accept(RoomListActions.ChangeRoomNotificationState(quickActions.roomId, RoomNotificationState.ALL_MESSAGES_NOISY))
+            }
+            is RoomListQuickActions.NotificationsAll          -> {
+                roomListViewModel.accept(RoomListActions.ChangeRoomNotificationState(quickActions.roomId, RoomNotificationState.ALL_MESSAGES))
+            }
+            is RoomListQuickActions.NotificationsMentionsOnly -> {
+                roomListViewModel.accept(RoomListActions.ChangeRoomNotificationState(quickActions.roomId, RoomNotificationState.MENTIONS_ONLY))
+            }
+            is RoomListQuickActions.NotificationsMute         -> {
+                roomListViewModel.accept(RoomListActions.ChangeRoomNotificationState(quickActions.roomId, RoomNotificationState.MUTE))
+            }
+            is RoomListQuickActions.Settings                  -> {
+                vectorBaseActivity.notImplemented("Opening room settings")
+            }
+            is RoomListQuickActions.Leave                     -> {
+                roomListViewModel.accept(RoomListActions.LeaveRoom(quickActions.roomId))
```

Comment:
Maybe ask for a confirmation with a dialog? Also I think there are some specific cases to handle (user is the last admin of the room, etc.)